### PR TITLE
Fix remaining spl and iterator callmaps

### PR DESCRIPTION
**Pull request by othercorey**

### CODE REVIEW

File: dictionaries/CallMap.php [hunk 1]
```diff
@@ -604,7 +604,7 @@
 'call_user_func_array' => ['mixed|false', 'callback'=>'callable', 'args'=>'list<mixed>'],
 'call_user_method' => ['mixed', 'method_name'=>'string', 'object'=>'object', 'parameter='=>'mixed', '...args='=>'mixed'],
 'call_user_method_array' => ['mixed', 'method_name'=>'string', 'object'=>'object', 'params'=>'list<mixed>'],
-'CallbackFilterIterator::__construct' => ['void', 'iterator'=>'Iterator', 'func'=>'callable(mixed):bool|callable(mixed,mixed):bool|callable(mixed,mixed,mixed):bool'],
+'CallbackFilterIterator::__construct' => ['void', 'iterator'=>'Iterator', 'callback'=>'callable(mixed):bool|callable(mixed,mixed):bool|callable(mixed,mixed,mixed):bool'],
```
Comment (othercorey) [inline dictionaries/CallMap.php:607]: Is this the correct/only way to have a callable with optional arguments?

Comment (orklah) [reply dictionaries/CallMap.php:607]: `callable(mixed,mixed=): bool` means "a callable with a mandatory first param then an optional second param"

Comment (othercorey) [reply dictionaries/CallMap.php:607]: I changed that callback to use 2 optional params instead.

**Comment by orklah** [comment]
Thanks!

**Pull request merged**